lib/fluent/plugin/out_file.rb

```diff
@@ -212,17 +213,27 @@ def write(chunk)
       FileUtils.mkdir_p File.dirname(path), mode: @dir_perm
 
       writer = case
-               when @compress_method.nil?
-                 method(:write_without_compression)
-               when @compress_method == :gzip
-                 if @buffer.compress != :gzip || @recompress
-                   method(:write_gzip_with_compression)
-                 else
-                   method(:write_gzip_from_gzipped_chunk)
-                 end
-               else
-                 raise "BUG: unknown compression method #{@compress_method}"
-               end
+                when @compress_method.nil?
+                  method(:write_without_compression)
+                when @compress_method == :gzip
+                  if @buffer.compress == :text || @recompress
+                    method(:write_with_compression).curry.call(@compress_method)
+                  elsif @buffer.compress == :gzip
+                    method(:write_from_compressed_chunk).curry.call(@compress_method)
+                  else
+                    raise Fluent::ConfigError, "You cannot specify different compression formats for Buffer (Buffer: #{@buffer.compress}, Self: #{@compress})"
```

Comment:
Thanks!
Could you make this `raise` run on `configure`?
It would be better to check the ConfigError at startup as `out_forward`.

Reply:
@daipom made the changes accordingly